ls1intum/Artemis
# `Development`: Improve code quality for exam

### Checklist
#### General



### Motivation and Context




### Description


- Add a DB migration for  `did_check_login`, `did_check_registration_number`, `did_check_registration_name` , `did_check_image` fields that fills with `false` (see [7783873b72fb5fcdb614e1643f8d30e3357fba34](https://github.com/ls1intum/Artemis/pull/6298/commits/7783873b72fb5fcdb614e1643f8d30e3357fba34))
- Better default values for exam user
- Apply suggestions left from this PR [#6137](https://github.com/ls1intum/Artemis/pull/6137)

Note: DB migration is approved by @krusche 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Course with exam (exam with different exercises)

1. Log in to Artemis
2. Navigate to Course Administration
3. Register students to exam 
4. Upload the pdf that contains student images, check if images are saved correctly.
5. Generate Student Exams (check if you can fetch all student exams and view specific student exam) 
6. Login as student 
7. Participate in exam and submit the exam
8. Log as instructor back and verify attendance (You should see students that started the exam but did not sign. ) 

Example files: 
[studentsWithImages.pdf](https://github.com/ls1intum/Artemis/files/10910361/studentsWithImages.pdf)
[studentsArtemis (1).csv](https://github.com/ls1intum/Artemis/files/10910367/studentsArtemis.1.csv)


### Review Progress






#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
